### DISCUSSION TITLE
Install to hard drive

### DISCUSSION THREAD

**Discussion by dm-vev** [CosmosOS/Cosmos#2366] [Q&A]
Can we install iso on hard drive? At least virtual, because on real equipment it is not very practical.

**Comment by selkij** [comment]
please read issues and wiki instead of always opening discussions

**Comment by dm-vev** [comment]
Perhaps I read it inattentively, but could you give a link to a page where there is an answer to my question?

**Comment by retrievaaaa** [comment]
The only thing I could imagine being possible is creating a new installer project, that just embeds all your OS files as resources and writes them to the disk

**Comment by AsertCreator** [comment]
i think its this scheme:
- install grub manually on drive
- copy kernel to drive
- copy resources to drive
it might sound simple, but its very hard. Firstly, you cant access attached iso because of poorly designed cosmos disk manager. Secondly, you cant format drive because of again cosmos disk manager. Thirdly, dont use Rufus-created drive as reference. It puts some other things and uses some sort of trampoline to actual iso boot sector located idkwhere. i tried to implement that but it didnt work

**Reply by tacitmarmot271** [reply]
YES AND YOU COULD use linux

**Comment by ADev531** [comment]
now just format and copy files, but i think sometimes format not works
